lxd/instance/drivers/driver_lxc.go

```diff
@@ -6567,6 +6568,40 @@ func (d *lxc) inheritInitPidFd() (int, *os.File) {
 	return -1, nil
 }
 
+type sftpProcess struct {
+	runningForkfile string
+	conn            net.Conn
+	cancel          *cancel.Canceller
+}
+
+var sftpMu sync.Mutex
+var sftpProcesses = make(map[string]*sftpProcess)
+
+// stop triggers the cancel function of the sftp process after a timeout.
+// If ctx doesn't have a deadline then a default timeout of 5s is added.
+func (p *sftpProcess) stop(ctx context.Context) error {
```

Comment:
this still does not trigger the stop until after the ctx is cancelled or timed out at 5s? 
I don't see why this is the case?

Reply:
From my understanding, `p.cancel.Cancel()` emits a cancellation event to the sftp process right ? Which is executed after `ctx` times out (default is 5s) no ? The following cancel signal being handled line 6815 in the select statement, which effectively kills the process, stops the real OS process and delete the 'sftpProcess' from the map.

Reply:
Sure, but you're not calling it until the supplied context has been cancelled. So its not doing an immediate a stop.
I'm not clear why its waiting for at least 5s before killing the process?

Compare what you have to what we have in the miniod process manager:

https://github.com/lxc/lxd/blob/master/lxd/storage/s3/miniod/miniod.go#L113-L116

Reply:
Oh ok I see what you mean. I'll remove that 5s second wait time then.